snapshots/windows/windows.go

```diff
@@ -175,18 +179,21 @@ func (s *snapshotter) Commit(ctx context.Context, name, key string, opts ...snap
 	}
 	defer t.Rollback()
 
-	usage := fs.Usage{
-		Size: 0,
+	// grab the existing id
+	id, _, _, err := storage.GetInfo(ctx, key)
+	if err != nil {
+		return err
 	}
 
-	if _, err = storage.CommitActive(ctx, key, name, snapshots.Usage(usage), opts...); err != nil {
-		return errors.Wrap(err, "failed to commit snapshot")
+	usage, err := fs.DiskUsage(ctx, s.getSnapshotDir(id))
+	if err != nil {
+		return err
 	}
 
-	if err := t.Commit(); err != nil {
-		return err
+	if _, err = storage.CommitActive(ctx, key, name, snapshots.Usage(usage), opts...); err != nil {
+		return errors.Wrap(err, "failed to commit snapshot")
 	}
-	return nil
+	return t.Commit()
```

Comment:
It looks like this is following what the overlay driver is doing, in that case maybe also update the rollback call
```
	defer func() {
		if err != nil {
			if rerr := t.Rollback(); rerr != nil {
				log.G(ctx).WithError(rerr).Warn("failed to rollback transaction")
			}
		}
	}()
```

Reply:
Done